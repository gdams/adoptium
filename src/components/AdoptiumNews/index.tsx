import React from 'react';
import moment from 'moment';
import { Trans } from 'gatsby-plugin-react-i18next';
import LinkText from '../LinkText'

const AdoptiumNews = () => {

    const adoptiumNews = {
        title: "Case Study: Building the World's Most Secure OpenJDK Distribution", 
        body: "Find out how the Eclipse Foundation and Adoptium Working Group are pioneering software supply chain security with Eclipse Temurin: <callToActionLink>Download now</callToActionLink>", 
        link: 'https://outreach.eclipse.foundation/adoptium-temurin-supply-chain-security?utm_campaign=Temurin%20Case%20Study&utm_source=website&utm_medium=adoptium%20docs', 
        // NOTE: Dates below are with the format: "YYYY-MM-dd"
        date: new Date('2024-05-01'),
        startDisplayAt: new Date('2024-05-15'),
        stopDisplayAfter: new Date('2024-05-31'),
    }

    const now = Date.now();
    if(!adoptiumNews || now < adoptiumNews.startDisplayAt.getTime() || now > adoptiumNews.stopDisplayAfter.getTime()) return;

    return (
        <div className='p-3 mt-4 mb-4 bg-light rounded-3 text-start'>
            <div className='container py-5'>
                <h2 className='text-pink'>{adoptiumNews.title}</h2>
                <div>
                    {adoptiumNews.date && <p className='m-0 fw-bold'>{moment(adoptiumNews.date).format('D MMMM YYYY')}</p>}
                    <p className='text-muted lh-sm'>
                        <Trans 
                            defaults={adoptiumNews.body} 
                            components={{
                                callToActionLink: <LinkText href={adoptiumNews.link||''} />
                            }}
                        />
                    </p>
                </div>
            </div>
        </div>
    );
};

export default AdoptiumNews;

export interface AdoptiumNewsItem {
    title: string;
    body: string;
    date?: Date;
    link?: string;
    startDisplayAt: Date;
    stopDisplayAfter: Date;
}
