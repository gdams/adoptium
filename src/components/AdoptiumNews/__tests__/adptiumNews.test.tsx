import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest'
import AdoptiumNews from '..';

describe('AdoptiumNews component', () => {
  it('renders correctly', () => {
    const { container } = render(<AdoptiumNews />);
    expect(container).toMatchSnapshot();
  });
});
