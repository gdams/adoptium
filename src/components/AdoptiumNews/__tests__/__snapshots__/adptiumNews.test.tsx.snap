// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AdoptiumNews component > renders correctly 1`] = `
<div>
  <div
    class="p-3 mt-4 mb-4 bg-light rounded-3 text-start"
  >
    <div
      class="container py-5"
    >
      <h2
        class="text-pink"
      >
        Case Study: Building the World's Most Secure OpenJDK Distribution
      </h2>
      <div>
        <p
          class="m-0 fw-bold"
        >
          1 May 2024
        </p>
        <p
          class="text-muted lh-sm"
        >
          Text
        </p>
      </div>
    </div>
  </div>
</div>
`;
